core/controller/src/main/scala/whisk/core/controller/WebActions.scala

```diff
@@ -264,37 +264,50 @@ protected[core] object WhiskWebActionsApi extends Directives {
   }
 
   /**
-   * Finds the content-type in the header list and maps it to a known media type. If it is not
-   * recognized, construct a failure with appropriate message.
+   * Finds the content-type in the header list and ensures that it is a valid format. If it is not
+   * valid, construct a failure with appropriate message.
+   * If the content-type header is missing, then return the supplied defaultType
    */
   private def findContentTypeInHeader(headers: List[RawHeader],
                                       transid: TransactionId,
                                       defaultType: MediaType): Try[MediaType] = {
     headers.find(_.lowercaseName == `Content-Type`.lowercaseName) match {
       case Some(header) =>
         MediaType.parse(header.value) match {
-          case Right(mediaType: MediaType) =>
-            // lookup the media type specified in the content header to see if it is a recognized type
-            MediaTypes.getForKey(mediaType.mainType -> mediaType.subType).map(Success(_)).getOrElse {
-              // this is a content-type that is not recognized, reject it
-              Failure(RejectRequest(BadRequest, Messages.httpUnknownContentType)(transid))
-            }
-          case _ => Failure(RejectRequest(BadRequest, Messages.httpUnknownContentType)(transid))
+          case Right(mediaType: MediaType) => Success(mediaType)
+          case _                           => Failure(RejectRequest(BadRequest, Messages.httpUnknownContentType)(transid))
         }
       case None => Success(defaultType)
     }
   }
 
+  private def isJsonFamily(mt: MediaType) = {
+    mt == `application/json` || mt.value.endsWith("+json")
+  }
+
   private def interpretHttpResponseAsJson(code: StatusCode,
                                           headers: List[RawHeader],
                                           js: JsValue,
                                           transid: TransactionId) = {
     findContentTypeInHeader(headers, transid, `application/json`) match {
-      case Success(mediaType) if (mediaType == `application/json`) =>
+      // use the default akka-http response marshaler for standard application/json
+      case Success(mediaType) if mediaType == `application/json` =>
         respondWithHeaders(removeContentTypeHeader(headers)) {
           complete(code, js)
         }
 
+      // for all other json-family content-type, explicitly marshal the response;
+      // the order of the case statement matters; isJsonFamily returns true for application/json
+      case Success(mediaType) if isJsonFamily(mediaType) =>
```

Comment:
`isJsonFamily()` checks for `application/json` here even though the type will never be `application/json` at this point.

Reply:
see my comment above - also this method is used elsewhere where both types are plausible.
"use the default akka-http response marshaler for standard application/json'

i debated collapsing both case matches but decided it was cleaner this way - i'm receptive to input.